Simplenote/Classes/SPSidebarContainerViewController.m

```diff
@@ -1,421 +1,461 @@
+{
+    return [self.sidebarView firstSubviewAsTableView];
+}
+
+- (UIViewController *)activeViewController
+{
+    return self.isSidebarVisible ? self.sidebarViewController : self.mainViewController;
+}
+
+- (UINavigationController *)mainNavigationController
+{
+    if (![self.mainViewController isKindOfClass:UINavigationController.class]) {
+        return nil;
     }
+
+    return (UINavigationController *)self.mainViewController;
 }
 
-- (void)setupForPanning {
-    
-    [self sidebarWillShow];
-    
-    if ([sidePanelViewDelegate respondsToSelector:@selector(containerViewControllerWillSlide:)])
-        [sidePanelViewDelegate containerViewControllerWillSlide:self];
-    
-    if ([sidePanelViewDelegate respondsToSelector:@selector(containerViewControllerWillShowSidePanel:)])
-        [sidePanelViewDelegate containerViewControllerWillShowSidePanel:self];
-    
-    static dispatch_once_t onceToken;
-    dispatch_once(&onceToken, ^{
-        sidePanelWidth = [self.theme floatForKey:@"containerViewSidePanelWidth"];
-    });
-    
-    CGRect sidePanelFrame = self.view.bounds;
-    sidePanelFrame.origin.x -= sidePanelWidth;
-    sidePanelFrame.size.width = sidePanelWidth;
-    _sidePanelViewController.view.frame = sidePanelFrame;
-    
-    dispatch_async(dispatch_get_main_queue(), ^{
-        [self.view insertSubview:self->_sidePanelViewController.view atIndex:0];
-    });
-    
-    _sidePanelViewController.view.autoresizingMask = UIViewAutoresizingFlexibleHeight | UIViewAutoresizingFlexibleRightMargin;
-    [self applySidePanelContentInsets];
-    
-    bSetupForPanning = YES;
+
+#pragma mark - Overridden Methods
+
+- (UIStatusBarStyle)preferredStatusBarStyle
+{
+    if (@available(iOS 13.0, *)) {
+        return UIStatusBarStyleDefault;
+    }
+
+    return SPUserInterface.isDark ? UIStatusBarStyleLightContent : UIStatusBarStyleDefault;
+}
+
+- (BOOL)shouldAutorotate
+{
+    return !self.isPanningActive && [self.activeViewController shouldAutorotate];
 }
 
-- (void)applySidePanelContentInsets {
 
-    UIEdgeInsets safeInsets = self.view.safeAreaInsets;
-    UIEdgeInsets contentInset = UIEdgeInsetsMake(safeInsets.top, 0, safeInsets.bottom, 0);
+#pragma mark - Initialization
 
-    [sidePanelViewDelegate containerViewController:self didChangeContentInset:contentInset];
+- (void)configureMainView
+{
+    self.view.backgroundColor = [UIColor colorWithName:UIColorNameBackgroundColor];
 }
 
-- (void)toggleSidePanel:(void (^)())completion {
-    
-    if (_bSidePanelVisible)
-        [self hideSidePanelAnimated:YES completion:completion];
-    else
-        [self showSidePanel:completion];
-    
+- (void)configurePanGestureRecognizer
+{
+    NSParameterAssert(self.mainView);
+
+    self.panGestureRecognizer = [[UIPanGestureRecognizer alloc] initWithTarget:self action:@selector(panGestureWasRecognized:)];
+    self.panGestureRecognizer.delegate = self;
+    [self.view addGestureRecognizer:self.panGestureRecognizer];
```

Comment:
Since we're modifying the view, maybe this should be called from `viewDidLoad` rather than `init`?

Reply:
Good call!! Nice catch, thank you!!